src/publish.proj

```diff
@@ -1,41 +1,66 @@
+    <ContainerName Condition="'$(__Container)' != '' and '$(PublishTestNativeBins)' != 'true'">$(__Container)</ContainerName>
+    <ContainerName Condition="'$(__Container)' != '' and '$(PublishTestNativeBins)' == 'true'">$(__Container)-$(__DistroRid)-$(__BuildArch)</ContainerName>
+    <ContainerName>$(ContainerName.Replace(".","-"))</ContainerName>
+    <AccountName>$(CloudDropAccountName)</AccountName>
+    <AccountKey>$(CloudDropAccessToken)</AccountKey>
   </PropertyGroup>
 
-  <Target Name="CreateContainerName"
-          DependsOnTargets="CreateVersionFileDuringBuild"
-          Condition="'$(ContainerName)' == '' or '$(PublishTestNativeBins)' == 'true'">
+  <Target Name="PublishPackages" Condition="'$(__PublishPackages)' == 'true'">
+    <PropertyGroup>
+      <FeedType>Blob</FeedType>
+      <RelativePath>$(__BuildType)/pkg</RelativePath>
+    </PropertyGroup>
+    <ItemGroup>
+      <ItemsToPush Remove="*.nupkg" />
+      <ItemsToPush Include="$(PackagesPattern)" Exclude="$(SymbolsPackagesPattern)"/>
+    </ItemGroup>
+    <PushToBlobFeed AccountName="$(AccountName)"
+                    AccountKey="$(AccountKey)"
+                    ContainerName="$(ContainerName)"
+                    ItemsToPush="@(ItemsToPush)"
+                    RelativePath="$(RelativePath)"
+                    PublishFlatContainer="$(PublishFlatContainer)"
+                    Overwrite="$(OverwriteOnPublish)" />
+  </Target>
+
+  <Target Name="PublishSymbolPackages" Condition="'$(__PublishSymbols)' == 'true'" >
     <PropertyGroup>
-      <ContainerName Condition="'$(__Container)' == '' and '$(PublishTestNativeBins)' != 'true'">coreclr-$(PreReleaseLabel)-$(BuildNumberMajor)-$(BuildNumberMinor)</ContainerName>
-      <ContainerName Condition="'$(__Container)' == '' and '$(PublishTestNativeBins)' == 'true'">coreclr-$(PreReleaseLabel)-$(BuildNumberMajor)-$(BuildNumberMinor)-$(__DistroRid)-$(__BuildArch)</ContainerName>
-      <ContainerName Condition="'$(__Container)' != '' and '$(PublishTestNativeBins)' != 'true'">$(__Container)</ContainerName>
-      <ContainerName Condition="'$(__Container)' != '' and '$(PublishTestNativeBins)' == 'true'">$(__Container)-$(__DistroRid)-$(__BuildArch)</ContainerName>
-      <ContainerName>$(ContainerName.Replace(".","-"))</ContainerName>
+      <FeedType>Blob</FeedType>
+      <RelativePath>$(__BuildType)/symbolpkg</RelativePath>
     </PropertyGroup>
+    <ItemGroup>
+      <ItemsToPush Remove="*.nupkg" />
+      <ItemsToPush Include="$(SymbolsPackagesPattern)"/>
+    </ItemGroup>
+    <PushToBlobFeed AccountName="$(AccountName)"
+                    AccountKey="$(AccountKey)"
+                    ContainerName="$(ContainerName)"
+                    ItemsToPush="@(ItemsToPush)"
+                    RelativePath="$(RelativePath)"
+                    PublishFlatContainer="$(PublishFlatContainer)"
```

Comment:
For these leg publishes shouldn't PublishFlatContainer be set to true?

Reply:
I actually don't know what that property does, is it documented somewhere?

Reply:
It is the difference between using a flat blob container or a blob feed. For these publishes I think they should be flat. For the final publish step it shouldn't be flat.

Reply:
PublishFlatContainer allows you to push to blob storage without creating a feed